queries/orderbook/order_rewards.sql

```diff
@@ -2,20 +2,21 @@ with trade_hashes as (SELECT solver,
                              order_uid,
                              fee_amount,
                              settlement.tx_hash,
-                             solver_competitions.id as auction_id
+                             auction_id
                       FROM trades t
                                LEFT OUTER JOIN LATERAL (
-                          SELECT tx_hash, solver
+                          SELECT tx_hash, solver, tx_nonce
                           FROM settlements s
                           WHERE s.block_number = t.block_number
                             AND s.log_index > t.log_index
                           ORDER BY s.log_index ASC
                           LIMIT 1
                           ) AS settlement ON true
-                               join solver_competitions
+                               join auction_transaction
                           -- This join also eliminates overlapping
                           -- trades & settlements between barn and prod DB
-                                    on settlement.tx_hash = solver_competitions.tx_hash
+                                   on settlement.solver = auction_transaction.tx_from
```

Comment:
This should be  using `settlement.tx_from`. I actually don't know why we have the solver column. Maybe from a time where we didn't use per-solver accounts?

Reply:
The solver column is part of the Settlement Event. I actually don't know why you have the `tx_from` column, since they are the same. solver was there first and should stay - perhaps you would like to reconsider adding the redundant column to the DB.

Reply:
Good point that it is part of the event. I thought we were manually inserting that in to the DB for some reason. In the contract code I see that the value of Settlement.solver is set to msg.sender . However, msg.sender is not the same as tx.origin [1]. For these purposes we want tx.origin because that is the address sending the full transaction and who's nonce got incremented with this transaction (correct me if I am wrong, please). So I would say that `tx_from` is more appropriate here.

[1] https://docs.soliditylang.org/en/develop/units-and-global-variables.html#block-and-transaction-properties

Reply:
From my understanding, `tx.origin` will always be `msg.sender` (or at least it always has been) partially because of the `onlySolver` modifier on the settlement method. Not sure if there is any plan for solvers to submit through:

```
modifier onlySolver {
        require(authenticator.isSolver(msg.sender), "GPv2: not a solver");
        _;
    }
```

Furthermore, the event is emitted using `msg.sender`:
```
        emit Settlement(msg.sender);
```

Since we use this event attribute for absolutely everything related to assigning a batch to a solver, I think we should remain consistent here in our payout scripts as well. 

Maybe @nlordell or @fedgiac have some ideas in mind of other contract integrations where `tx.origin` might be different from `msg.sender` or possibly where the event might emit a different value that we are actually expecting.

Reply:
```suggestion
                                   on settlement.tx_from = auction_transaction.tx_from
```

Reply:
I don't think the values have been different in the past but yes I'm saying theoretically this could happen. At the moment all our settlements are transactions submitted directly to the settlement contract from EOAs but you could imagine that another smart contract is a solver and a settlement would go through it. I think this is was Nick did in https://github.com/cowprotocol/services/pull/599 .

Reply:
I just hope that these kind of special settlement actually emit the correct solver in that case.

Reply:
Am unresolving so Nick and Federico can see why they got pinged. Maybe the have input on this.

Reply:
In general `tx.origin != msg.sender` if and only if the solver that executes the settlement is (at the time) a smart contract. This is currently the case only for our team multisig on mainnet, but I believe we'll expand on this at some point.

Note also that in the case of a smart contract solver it's possible to have two settlement in the same transaction (same nonce, origin, and hash).

Reply:
Thanks for the insight @fedgiac. Do you have any suggestions as to how we might overcome the hurdle of non-uniqueness of the pair `(tx_from, tx_nonce)`. I suspect there is currently a uniqueness constraint on the DB level.

Reply:
I noticed this when designing the DB schema for this feature and decided it was not worth trying to support multiple settlements in the same transaction yet. So is is known. The are some ways to do it while supporting that but haven't thought about it in detail because it is unclear if we will do this.